test_opensearchpy/test_connection.py

```diff
@@ -284,6 +286,61 @@ def test_http_auth_list(self):
             con.headers,
         )
 
+    @pytest.mark.skipif(
+        sys.version_info < (3, 6), reason="AWSV4SignerAuth requires python3.6+"
+    )
+    def test_aws_signer_as_http_auth(self):
+        dummy_session = self.fetch_dummy_session()
+        region = "us-west-2"
+
+        import subprocess
+
+        from opensearchpy.helpers.signer import AWSV4SignerAuth
+
+        subprocess.check_call([sys.executable, "-m", "pip", "install", "botocore"])
```

Comment:
why is this required?

Reply:
Not required. It is unit test. We can validate only correctness

Reply:
So for tests should I add botocore as the dependency and remove pip install and uninstall from the test cases?

Reply:
Yes, you should add botocore as a dev dependency if you need it for tests.

Reply:
removed installation of `botocore` from the test case and moved it to `setup.py`.